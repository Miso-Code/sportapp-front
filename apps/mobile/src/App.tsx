import React from 'react'

import { SafeAreaView } from 'react-native'
import { Colors } from 'react-native/Libraries/NewAppScreen'
import TestComponent from './TestComponent'

function App(): React.JSX.Element {
	const backgroundStyle = {
		backgroundColor: Colors.lighter
	}

	return (
		<SafeAreaView style={backgroundStyle}>
			<TestComponent />
		</SafeAreaView>
	)
}

export default App
