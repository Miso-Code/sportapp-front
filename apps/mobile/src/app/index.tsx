import React, { useEffect } from 'react'
import { View, StyleSheet } from 'react-native'
import { ActivityIndicator } from 'react-native-paper'
import { router, useRootNavigationState } from 'expo-router'
import { useAuthStore } from '@sportapp/stores'

const Index: React.FC = () => {
	const { isAuth } = useAuthStore()
	const rootNavigationState = useRootNavigationState()

	useEffect(() => {
		if (rootNavigationState?.key)
			router.navigate(isAuth ? 'profile' : 'login')
	}, [isAuth, rootNavigationState?.key])

	return (
		<View style={styles.container}>
			<ActivityIndicator
				animating={true}
				size='large'
				testID='progressBar'
			/>
		</View>
	)
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: 'center',
		alignItems: 'center'
	}
})

export default Index
