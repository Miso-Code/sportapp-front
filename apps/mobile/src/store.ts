import { deleteItemAsync, getItemAsync, setItemAsync } from 'expo-secure-store'

import { StateStorage } from 'zustand/middleware'

export const expoSecureStorage: StateStorage = {
	getItem: async (name: string): Promise<string | null> => {
		return (await getItemAsync(name)) || null
	},
	setItem: async (name: string, value: string): Promise<void> => {
		await setItemAsync(name, value)
	},
	removeItem: async (name: string): Promise<void> => {
		await deleteItemAsync(name)
	}
}
