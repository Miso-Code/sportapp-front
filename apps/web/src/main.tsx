import '@/index.scss'
import Router from '@/router'
import React from 'react'
import ReactDOM from 'react-dom/client'
import '../config/lang/form'
import '../config/lang/i18n'
import { StyledEngineProvider } from '@mui/material'
import { ThemeProvider, createTheme } from '@mui/material/styles'

const theme = createTheme({
	palette: {
		primary: {
			light: '#677ea7',
			main: '#425E91',
			dark: '#2e4165',
			contrastText: '#fff'
		}
	}
})

ReactDOM.createRoot(document.getElementById('root')!).render(
	<React.StrictMode>
		<ThemeProvider theme={theme}>
			<StyledEngineProvider injectFirst>
				<Router />
			</StyledEngineProvider>
		</ThemeProvider>
	</React.StrictMode>
)
