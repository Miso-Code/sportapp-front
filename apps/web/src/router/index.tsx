import PrivateRoutes from '@/router/private/routes'
import PublicRoutes from '@/router/public/routes'
import { useEffect } from 'react'
import { useTranslation } from 'react-i18next'
import { createBrowserRouter, RouterProvider } from 'react-router-dom'

const Router = () => {
	const router = createBrowserRouter([PrivateRoutes(), ...PublicRoutes()])

	const { i18n } = useTranslation()

	useEffect(() => {
		document.documentElement.lang = i18n.language
	}, [i18n.language])

	return <RouterProvider router={router} />
}

export default Router
