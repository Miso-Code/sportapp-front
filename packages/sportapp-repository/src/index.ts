import axios, { AxiosInstance } from 'axios'
import { globalVariables } from './utils/global-variables'

export const sportappApi: AxiosInstance = axios.create({
	baseURL: globalVariables().EXPO_PUBLIC_SPORTAPP_API_URL,
	headers: {}
})
