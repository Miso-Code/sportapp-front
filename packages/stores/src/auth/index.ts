import UserApi from '@sportapp/sportapp-repository/src/user'
import {
	RegisterFullUserRequest,
	RegisterUserRequest
} from '@sportapp/sportapp-repository/src/user/interfaces'
import { mountStoreDevtool } from 'simple-zustand-devtools'
import { create } from 'zustand'
import { createJSONStorage, persist } from 'zustand/middleware'
import { IAuthState, IAuthStore, User } from './interfaces'

export const initialAuthState: IAuthState = {
	user: undefined,
	isAuth: false,
	error: undefined,
	loading: false
}

export const useAuthStore = create(
	persist<IAuthStore>(
		(set, get) => ({
			...initialAuthState,
			login: async (email, password) => {
				// WIP login logic
				if (email === 'a' && password === 'b') {
					set((state) => ({ ...state, isAuth: true }))
					return true
				}
				return false
			},
			logout: () => {
				// WIP logout logic
				set((state) => ({
					...state,
					user: undefined,
					isAuth: false,
					loading: false,
					error: undefined
				}))
			},
			register: async (request: RegisterUserRequest) => {
				const userApi = new UserApi()
				try {
					set((state) => ({
						...state,
						loading: true
					}))
					const response = await userApi.register(request)

					if (
						response &&
						typeof response === 'object' &&
						response.id
					) {
						const userPayload: User = {
							...response
						}
						set((state) => ({
							...state,
							user: userPayload,
							loading: false
						}))
						return true
					}

					set((state) => ({
						...state,
						error: 'errors.register.base',
						loading: false
					}))

					return false
				} catch (e) {
					set((state) => ({
						...state,
						loading: false,
						error: 'errors.register.base'
					}))
					return false
				}
			},
			registerFull: async (request: RegisterFullUserRequest) => {
				const userApi = new UserApi()
				const user = get().user

				if (!user) {
					return false
				}

				set((state) => ({
					...state,
					loading: true,
					isAuth: true
				}))

				return await userApi.registerFull(user.id, request)
			},
			setError: (error) => set({ error }),
			setLoading: (loading) => set({ loading }),
			setUser: (user: User) => set({ user }),
			clearState: () =>
				set((state) => ({ ...state, ...initialAuthState }))
		}),
		{
			name: 'auth-storage',
			storage: createJSONStorage(() => sessionStorage)
		}
	)
)

if (process.env.NODE_ENV === 'development') {
	mountStoreDevtool('AuthStore', useAuthStore)
}
