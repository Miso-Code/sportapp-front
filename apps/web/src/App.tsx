import { useTranslation } from 'react-i18next'
import './App.scss'

function App() {
	const { t } = useTranslation()

	return (
		<>
			<h1>{t('app.name')}</h1>
		</>
	)
}

export default App
