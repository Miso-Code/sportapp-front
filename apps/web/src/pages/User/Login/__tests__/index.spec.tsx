import { useAuthStore } from '@sportapp/stores/src/auth'
import {
	RenderResult,
	fireEvent,
	render,
	screen,
	waitFor
} from '@testing-library/react'
import { useState } from 'react'
import { act } from 'react-dom/test-utils'
import { useNavigate } from 'react-router-dom'
import LoginPage from '../index'

jest.mock(
	'@/containers/Login',
	() =>
		({
			onHandleSubmit
		}: {
			onHandleSubmit: (data: { email: string; password: string }) => void
		}) =>
			(
				<button
					data-testid='login-button'
					onClick={() => {
						onHandleSubmit({ email: 'test', password: 'test' })
					}}>
					Login
				</button>
			)
)

jest.mock('@sportapp/stores/src/auth', () => ({
	useAuthStore: jest.fn().mockReturnValue({
		login: jest.fn().mockReturnValue(true),
		logout: jest.fn(),
		loading: false,
		error: null,
		isAuth: false
	})
}))

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useNavigate: jest.fn()
}))

jest.mock('react', () => {
	const ActualReact = jest.requireActual('react')
	return {
		...ActualReact,
		useState: jest.fn(() => [false, jest.fn()])
	}
})

jest.mock(
	'@/components/TransitionAlert',
	() =>
		({ isOpen }: { isOpen: boolean }) =>
			isOpen ? <div data-testid='alert'>Alert</div> : null
)

describe('LoginPage', () => {
	let wrapper: RenderResult

	beforeEach(() => {
		wrapper = render(<LoginPage />)
	})

	afterEach(() => {
		jest.clearAllMocks()
		wrapper.unmount()
	})

	it('should render the component', () => {
		expect(wrapper.container).toMatchSnapshot()
	})

	it('should render the title', () => {
		expect(screen.getByText('app.name')).toBeInTheDocument()
	})

	it('should render the card title', () => {
		expect(screen.getByText('login.default')).toBeInTheDocument()
	})

	it('should render the login button', () => {
		expect(screen.getByTestId('login-button')).toBeInTheDocument()
	})

	it('should call navigate when the form is submitted', async () => {
		const navigate = jest.fn()
		;(useNavigate as jest.Mock).mockReturnValue(navigate)

		wrapper.rerender(<LoginPage />)

		const button = wrapper.getByTestId('login-button')

		fireEvent.click(button)

		await waitFor(() => expect(navigate).toHaveBeenCalledWith('/home'))
	})

	it('should call setAlert when the form is submitted', async () => {
		const setAlert = jest.fn()
		;(useState as jest.Mock).mockImplementationOnce(() => [false, setAlert])
		;(useAuthStore as unknown as jest.Mock).mockReturnValue({
			login: jest.fn().mockReturnValue(false),
			loading: false,
			logout: jest.fn(),
			error: null,
			isAuth: false
		})

		wrapper.rerender(<LoginPage />)

		const button = wrapper.getByTestId('login-button')

		act(() => {
			fireEvent.click(button)
		})

		await waitFor(() => expect(setAlert).toHaveBeenCalledWith(true))
	})

	it('should called navigate to register page', async () => {
		const navigate = jest.fn()
		;(useNavigate as jest.Mock).mockReturnValue(navigate)

		wrapper.rerender(<LoginPage />)

		const button = screen.getByText('login.register.go')

		fireEvent.click(button)

		await waitFor(() => expect(navigate).toHaveBeenCalledWith('/register'))
	})
})
