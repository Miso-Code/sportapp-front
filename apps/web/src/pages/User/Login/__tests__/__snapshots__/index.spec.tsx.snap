// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should render the component 1`] = `
<div>
  <div
    class="login"
  >
    <main
      class="section-main"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 title css-o2w69a-MuiTypography-root"
      >
        app.name
      </h1>
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded card-login css-172kgz0-MuiPaper-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 card-title css-2ulfj5-MuiTypography-root"
        >
          login.default
        </h6>
        <button
          data-testid="login-button"
        >
          Login
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-fullWidth navigation css-10vjzfb-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          title="login.button"
          type="button"
        >
          login.register.go
        </button>
      </div>
      <div
        class="navigation-partner MuiBox-root css-0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-sbn2d4-MuiTypography-root"
        >
          login.partner.question
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            title="login.button"
            type="button"
          >
            login.partner.go
          </button>
        </p>
      </div>
    </main>
    <section
      class="section-secondary"
    >
      <figure
        class="container-image"
      >
        <img
          alt="login.image"
          class="image-register"
          src="mocked-file-content"
        />
      </figure>
    </section>
  </div>
</div>
`;
